app/controllers/api/base_controller/generic.rb

```diff
@@ -152,6 +153,15 @@ def validate_type(klass, type)
         raise BadRequestError, "Invalid type #{type} specified - #{err}"
       end
 
+      def subcollection_data(type, data)
+        collection_config.subcollections(type).each_with_object({}) do |sc, hash|
+          if data.key?(sc.to_s)
+            hash[sc] = data[sc.to_s]
+            data.delete(sc.to_s)
+          end
+        end
+      end
```

Comment:
Is this all just basically `Hash#extract!`?
```suggestion
      def subcollection_data(type, data)
        data.extract!(collection_config.subcollections(type).map(&:to_s))
      end
```

Reply:
Not going to exit this because this was me just extracting this out of the previous method to share it.

https://github.com/ManageIQ/manageiq-api/pull/916/files#diff-6388f564e8b184b4d6d0380cfd06f502fff6fe45fbd1972cf5c66a47857912d4L16-L21

Refactoring like this can come at a later date.